src/services/city_services/update_city.js

```diff
@@ -0,0 +1,21 @@
+class UpdateCity {
+    constructor({ cityDaos }) {
+        this.cityDaos = cityDaos;
+
+        this.execute = this.execute.bind(this);
+    }
+    async execute(params) {
+        try {
+            const { id, titles,thumnail,is_pinned } = params;
+            const daosResult = await this.cityDaos.update(id, titles,thumnail,is_pinned);
+            //I change to this because "Cannot read property 'failure' of null"
```

Comment:
dòng comment này không có ý nghĩa. M bị lỗi `"Cannot read property 'failure' of null"` là do trong `daos` m `return updatedCity` thay vì `return { updatedCity }`, tức là lỗi đó là do m không consistent trong cách code chứ không phải vì lý do khách quan nào khác.

Cụ thể: nếu m `return { updatedCity }` như những lần trước, thì ngay cả khi 
`updatedCity == null`, `daosResult.failure` sẽ trả về undefined chứ sẽ không raise lỗi như m comment

Ở đây t không nói cách code hiện tại của m là sai, t handle logic đúng, chỉ khác coding style, nhưng ý t là comment chỉ nên dùng để `giải thích những những đoạn code có thể gây confuse` chứ không nên dùng để `giải thích những gì m đã làm`